### ISSUE TITLE
Fix violation position after mixin

### ISSUE THREAD

**Issue by MLefebvreICO** [stylelint/stylelint#3509]
> Clearly describe the bug

Error positioning is wrong when a mixin is used before the error scope, putting the rule's error on the wrong line (seems to depends on the size of the mixing function line numbers)

> Which rule, if any, is the bug related to?

All.

> What CSS is needed to reproduce the bug?

A mixing `is-mobile` : 
```less
.is-mobile(@rules) {
    @media screen and (max-width: 1200px) {
        @rules();
    }
}
``` 
The following code : 

```css
.something {
    color: black;

    .is-mobile({
        color: red;
        background-color: black;
    })
}

.foo {
    height: 20px;
    width: 40px;
      background-color: pink;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "indentation": 4
  }
}
```

> Which version of stylelint are you using?

`9.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

* CLI with `stylelint stylesheets/** --config ../../.stylelintrc`
* VSCode plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes it is related to LESS mixins.

> What did you expect to happen?

```shell
someFile.less
13:7   ✖  Expected indentation of 4 spaces (indentation)
```

> What actually happened (e.g. what warnings or errors did you get)?

```shell
someFile.less
10:7   ✖  Expected indentation of 4 spaces (indentation)
```

**Comment by MLefebvreICO** [comment]
Real case scenario in our codebase with another rule : 
![image](https://user-images.githubusercontent.com/17457252/43589966-dcb3340a-963d-11e8-8bce-858266a74346.png)

**Comment by jeddy3** [comment]
This has been labelled as "help wanted". As this is an open source project, reliant on people's free time, please consider helping to investigate a fix. As @evilebottnawi mentioned in https://github.com/stylelint/stylelint/issues/3511#issuecomment-410320888, this might be issue upstream in the Less parser. Helping to narrow down where the problem lies would be a great first step.

**Comment by jwilsson** [comment]
I think this is fixed with `stylelint@9.7.1` using the new Less parser.

**Comment by jeddy3** [comment]
@jwilsson Thanks good news! I'll close for now as we can always reopen if it reoccurs.
